.travis.yml

```diff
@@ -0,0 +1,62 @@
+sudo: required
+
+dist: xenial
+
+language: java
+
+services:
+  - docker
+
+env:
+  - JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-amd64
+
+# Run before every job
+before_install:
+  - sudo apt-get update
+  - sudo apt-get install -y openjdk-8-jdk
+  - PATH=$(echo "$PATH" | sed -e 's/:\/usr\/local\/lib\/jvm\/openjdk11\/bin//')
+  - java -version
+  - uname -a
+  - chmod +x pom.xml
+
+# Only running report_coverage and deploy stage (after merge) to master branch
+stages:
+  - test
+  - name: report_coverage
+    if: branch = master AND type != pull_request
+  - name: deploy
+    if: branch = master AND type != pull_request
+
+# List of jobs to run, tied to specific stages
+jobs:
+  include:
+    - stage: test
+      name: unitAndFunctionalTests
+      script:
+        - mvn -Dsurefire.rerunFailingTestsCount=2 test -DskipITs -B
```

Comment:
Why can't we start off with a simple mvn test until we actually write the tests and understand what kind of flags we need to add to it later?

Reply:
simplified